qiskit/algorithms/optimizers/steppable_optimizer.py

```diff
@@ -0,0 +1,286 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""SteppableOptimizer interface"""
+
+from abc import abstractmethod, ABC
+from dataclasses import dataclass
+from typing import Union, Callable, Optional, Tuple, List
+from .optimizer import Optimizer, POINT, OptimizerResult
+
+CALLBACK = Callable
```

Comment:
Does this have any value/is needed?

Reply:
Right now the callbacks for every optimizer are different. All they have in common is that they are always Callable. I guess I could actually do ``Callable[Any,None]``